test/dd/test_dd_noise_functionality.cpp

```diff
@@ -90,44 +90,101 @@ TEST_F(DDNoiseFunctionalityTest, DetSimulateAdder4TrackAPD) {
-      auto dd = std::make_unique<DensityMatrixTestPackage>(qc.getNqubits());
-
-      auto rootEdge = dd->makeZeroDensityOperator(qc.getNqubits());
-      dd->incRef(rootEdge);
-
-      const auto noiseEffects = {dd::AmplitudeDamping, dd::PhaseFlip,
-                                 dd::Depolarization, dd::Identity};
-
-      auto deterministicNoiseFunctionality =
-          dd::DeterministicNoiseFunctionality(
-              dd, qc.getNqubits(), 0.01, 0.02, 0.02, 0.04, noiseEffects,
-              useDensityMatrixType, applyNoiseSequentially);
-
-      for (auto const& op : qc) {
-        dd->applyOperationToDensity(rootEdge, dd::getDD(op.get(), *dd),
-                                    useDensityMatrixType);
-        deterministicNoiseFunctionality.applyNoiseEffects(rootEdge, op);
-      }
+  const std::array<std::array<dd::SparsePVecStrKeys, 2>, 2> results{};
+  auto dd = std::make_unique<DensityMatrixTestPackage>(qc.getNqubits());
 
-      const auto m = rootEdge.getSparseProbabilityVectorStrKeys(0.001);
-      results[static_cast<std::size_t>(useDensityMatrixType)]
-             [static_cast<std::size_t>(applyNoiseSequentially)] = m;
-    }
+  auto rootEdge = dd->makeZeroDensityOperator(qc.getNqubits());
+  dd->incRef(rootEdge);
+
+  const auto noiseEffects = {dd::AmplitudeDamping, dd::PhaseFlip,
+                             dd::Depolarization, dd::Identity};
+
+  auto deterministicNoiseFunctionality = dd::DeterministicNoiseFunctionality(
+      dd, qc.getNqubits(), 0.01, 0.02, 0.02, 0.04, noiseEffects);
+
+  for (auto const& op : qc) {
+    dd->applyOperationToDensity(rootEdge, dd::getDD(op.get(), *dd));
+    deterministicNoiseFunctionality.applyNoiseEffects(rootEdge, op);
   }
+
+  const auto m = rootEdge.getSparseProbabilityVectorStrKeys(0.001);
+
   // Expect that all results are the same
   static constexpr fp TOLERANCE = 1e-10;
-  for (auto& result : results) {
-    for (auto& j : result) {
+  for (const auto& result : results) {
+    for (const auto& j : result) {
       for (const auto& [key, value] : j) {
         EXPECT_NEAR(value, reference.at(key), TOLERANCE);
       }
     }
   }
 }
 
+TEST_F(DDNoiseFunctionalityTest, testingMeasure) {
+  std::mt19937_64 mt{0}; // NOLINT(cert-msc51-cpp)
+
+  qc::QuantumComputation qcOp{};
+
+  qcOp.addQubitRegister(3U);
+  qcOp.h(0);
+  qcOp.h(1);
+  qcOp.h(2);
+
+  auto dd = std::make_unique<DensityMatrixTestPackage>(qcOp.getNqubits());
+
+  auto rootEdge = dd->makeZeroDensityOperator(qcOp.getNqubits());
+  dd->incRef(rootEdge);
+
+  auto deterministicNoiseFunctionality = dd::DeterministicNoiseFunctionality(
+      dd, qcOp.getNqubits(), 0.01, 0.02, 0.02, 0.04, {});
+
+  for (auto const& op : qcOp) {
+    dd->applyOperationToDensity(rootEdge, dd::getDD(op.get(), *dd));
+    deterministicNoiseFunctionality.applyNoiseEffects(rootEdge, op);
+  }
+  char result = '2';
+
+  const double tolerance = 1e-10;
+
+  auto tmp = rootEdge.getSparseProbabilityVectorStrKeys();
+  auto prob = 1.0 / 8;
```

Comment:
```suggestion
  auto prob = 0.125;
```
Just because it's so easy to write down exactly.

Reply:
I changed the code as suggested.